signer/xpi/jar.go

```diff
@@ -86,13 +140,20 @@ func makeJARManifest(input []byte) (manifest []byte, err error) {
 		if err != nil {
 			return manifest, err
 		}
-		fmt.Fprintf(mw, "Name: %s\nDigest-Algorithms: SHA1 SHA256\n", f.Name)
+
+		filename, err := formatFilename([]byte(f.Name))
+		if err != nil {
+			return manifest, err
+		}
 		h1 := sha1.New()
 		h1.Write(data)
-		fmt.Fprintf(mw, "SHA1-Digest: %s\n", base64.StdEncoding.EncodeToString(h1.Sum(nil)))
 		h2 := sha256.New()
 		h2.Write(data)
-		fmt.Fprintf(mw, "SHA256-Digest: %s\n\n", base64.StdEncoding.EncodeToString(h2.Sum(nil)))
+
+		fmt.Fprintf(mw, "Name: %s\nDigest-Algorithms: SHA1 SHA256\nSHA1-Digest: %s\nSHA256-Digest: %s\n\n",
```

Comment:
Noticed that the apk jar packer uses `SHA-256` instead of `SHA256` and also doesn't specify digest-algorithms: https://github.com/mozilla-services/autograph/blob/master/signer/apk/jar.go#L103

Reply:
yes, xpi and apk use different manifest syntaxes

Reply:
:+1: ok de-duping this might not be the best idea then.